test/circuit_optimizer/test_collect_blocks.cpp

```diff
@@ -16,15 +16,31 @@
 namespace qc {
 TEST(CollectBlocks, emptyCircuit) {
   QuantumComputation qc(1);
-  qc::CircuitOptimizer::collectBlocks(qc, 1);
+  qc::CircuitOptimizer::collectBlocks(qc, 1, false);
+  EXPECT_EQ(qc.getNindividualOps(), 0);
+}
+
+TEST(CollectBlocks, emptyCliffordCircuit) {
+  QuantumComputation qc(1);
+  qc::CircuitOptimizer::collectBlocks(qc, 1, true);
   EXPECT_EQ(qc.getNindividualOps(), 0);
 }
 
 TEST(CollectBlocks, singleGate) {
   QuantumComputation qc(1);
   qc.h(0);
   std::cout << qc << "\n";
-  qc::CircuitOptimizer::collectBlocks(qc, 1);
+  qc::CircuitOptimizer::collectBlocks(qc, 1, false);
+  std::cout << qc << "\n";
+  EXPECT_EQ(qc.size(), 1);
+  EXPECT_TRUE(qc.front()->isStandardOperation());
+}
+
+TEST(CollectBlocks, singleCliffordGate) {
+  QuantumComputation qc(1);
+  qc.t(0);
+  std::cout << qc << "\n";
+  qc::CircuitOptimizer::collectBlocks(qc, 1, true);
```

Comment:
The T gate is not a Clifford gate. This seems to be a non-sense test. Please adjust to make this a reasonable test.